pennychat/views.py

```diff
@@ -1,3 +1,77 @@
-# from django.shortcuts import render
+from rest_framework import viewsets, mixins, generics
+from rest_framework.reverse import reverse
+from rest_framework.response import Response
+from rest_framework.status import HTTP_201_CREATED, HTTP_400_BAD_REQUEST
 
-# Create your views here.
+from users.models import UserProfile
+from .models import PennyChat, FollowUp
+from .serializers import PennyChatSerializer, FollowUpSerializer
+
+
+class PennyChatViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows users to be viewed or edited.
+    """
+    queryset = PennyChat.objects.all().order_by('-date')
+    serializer_class = PennyChatSerializer
+
+    def perform_create(self, serializer):
+        # get or create an anonymous user since we don't have auth
+        # this will eventually be replaced by user = self.request.user
+        user, created = UserProfile.objects.get_or_create(
+            real_name='Anonymous Profile',
+            defaults={
+                'email': 'anonymous@profile.com'
+            })
+        serializer.save(user=user)
```

Comment:
This confuses me. It seems to save a penny chat with no content and it attaches it to an anonymous user. I don't see how that will ever be useful. I'm not going to block shipping on this, it just represents something that I don't understand yet.

Reply:
`perform_create` is called in the background from a post in a ModelViewSet. The serializer object here will have all the relevant data that is needed for the chat. I attach the anonymous user for now since there is no other way of figuring out who is created the chat without auth.